app/models/manageiq/providers/openstack/manager_mixin.rb

```diff
@@ -495,5 +546,14 @@ def orchestration_template_validate(template)
     raise MiqException::MiqOrchestrationValidationError, err.to_s, err.backtrace
   end
 
+  def authentication_class(authtype)
```

Comment:
Can this be in the `InfraManager` instead of the `ManagerMixin`?  That way it isn't accidentally returning `ManageIQ::Providers::Openstack::InfraManager::AuthKeyPair` for the Openstack Cloud/Network/Storage managers